controller/gke-cluster-config-handler.go

```diff
@@ -665,6 +665,12 @@ func BuildUpstreamClusterState(cluster *gkeapi.Cluster) (*gkev1.GKEClusterConfig
 		newSpec.MaintenanceWindow = &cluster.MaintenancePolicy.Window.DailyMaintenanceWindow.StartTime
 	}
 
+	autopilot := false
+	if cluster.Autopilot != nil && cluster.Autopilot.Enabled {
+		autopilot = true
+	}
+	newSpec.Autopilot = &autopilot
+
```

Comment:
If we are exposing this option in the UI we need to watch out for immutability, meaning disable on update. We also need to communicate the details of create, like if you select it then zone must go away and scalers get turned on and input disabled.

Reply:
Good point, I will make a note of that in the UI issue. This is not the only attribute that can't be updated, so I'll try to document them all.

Reply:
It seems like we should also be checking that logic here. For instance if I create a cluster via CR, and I use zone, what happens ?

Reply:
For immutability you can add the norman tag "noupdate" in gke-operator and rancher will respect it, but this simply cause it to be removed from requests instead of informing the user that it cannot be updated via an error.

Reply:
Oh cool, that's already being used, nvm.

Reply:
> It seems like we should also be checking that logic here.

We've already opted not to do heavy validation here for most settings and instead lean on the UI and GKE to validate.

Reply:
As long as error message gets bubbled up clearly that should work.